packages/lexical/src/LexicalSelection.ts

```diff
@@ -822,7 +822,7 @@ export class RangeSelection implements BaseSelection {
       }
     }
 
-    if (selectedNodesLength === 1) {
+    if (anchor.key === focus.key && selectedNodesLength === 1) {
```

Comment:
Nit, resolving properties at the end of the condition
```suggestion
    if (selectedNodesLength === 1 && anchor.key === focus.key) {
```